lighthouse-core/lib/dependency-graph/simulator/connection-pool.js

```diff
@@ -139,14 +139,10 @@ module.exports = class ConnectionPool {
    * @return {?TcpConnection}
    */
   acquire(record, options = {}) {
-    if (this._connectionsByRecord.has(record)) {
-      // @ts-ignore
-      return this._connectionsByRecord.get(record);
-    }
+    if (this._connectionsByRecord.has(record)) throw new Error('Record already has a connection');
 
-    const origin = String(record.parsedURL.securityOrigin);
+    const origin = record.parsedURL.securityOrigin;
     const observedConnectionWasReused = !!this._connectionReusedByRequestId.get(record.requestId);
-    /** @type {TcpConnection[]} */
```

Comment:
this isn't necessary anymore?

Reply:
> this isn't necessary anymore?

no, its inferred from the first part of the init and the `_findAvailableConnectionWithLargestCongestionWindow` callsite